homeassistant/components/modbus/sensor.py

```diff
@@ -141,10 +145,12 @@ async def async_setup_platform(
             _LOGGER.error("Error in sensor %s structure: %s", entry[CONF_NAME], err)
             continue
 
-        if entry[CONF_COUNT] * 2 != size:
+        if regSize != size:
             _LOGGER.error(
-                "Structure size (%d bytes) mismatch registers count (%d words)",
+                "Structure size (%d bytes) mismatch registers size (%d bytes). Multi: %d, Count: %d",
```

Comment:
Please you complicate something simple...why have both a register size which in your case is fix 2 and an multiplier...it is a lot easier and more understandable to declare the size of a register (with padding etc).